### ISSUE TITLE
difference of loss curve btw llama and mamba supervised fine-tuning

### ISSUE THREAD

**Issue by ftgreat** [state-spaces/mamba#125]
I have pre-trained  the same 300B tokens from scratch for both mamba and llama models (with tie embeddings). And main metrics show similar evaluation results. However, when conducting supervised fine-tuning (SFT) on the same instruct dataset, two models exhibit big differences:

1. Initial training loss: Mamba starts with a higher loss, similar to the beginning of pretraining (around 11.x+), while Llama's loss is closer to the end of its pretraining phase (about 2.x).
2. Training curve: Mamba suddenly decreases from a higher start loss to a small loss, whereas Llama's loss decreases much slowly. At epoch=3, Mamba's loss is significantly lower than Llama's. However, main metrics at epoch 3 show similar evaluation results, including human evaluation results.

What might be the reason for these observations? Could it be related to Mamba's model structure, or is it an issue with the hyperparameter settings? Thank you. @tridao 

P.S.: We have tried Llama's across various model sizes in SFT training, and the llama's observations(training loss & eval loss) appear similar.

![image](https://github.com/state-spaces/mamba/assets/10432020/dfc68575-a184-4e3b-904d-c7f19431b0b8)

**Comment by tridao** [comment]
That seems strange to me, but I haven't run SFT on Mamba. Maybe there's some issue with padding / masking?

**Comment by ftgreat** [comment]
> That seems strange to me, but I haven't run SFT on Mamba. Maybe there's some issue with padding / masking?

indeed there was a bug. Thanks.
